app/jobs/split_pdf_job.rb

```diff
@@ -298,9 +270,21 @@ def save_pages(exam_template, partial_exams, filename=nil, split_pdf_log=nil)
     num_complete
   end
 
-  # TODO: add in matching using name/username
-  def match_student(_first_name, _last_name, student_id, _username, _exam)
-    Student.find_by(id_number: student_id)
+  # Determine a match using the parsed handwritten text and the user identifying field (+exam_template.cover_fields+).
+  # If the parsing was successful, +parsed+ is a list of two strings parsed from the handwritten text:
+  # the first is the result of attempting to parse alphabetic characters, and the second is the result of
+  # attempting to parse numeric digits.
+  def match_student(parsed, exam_template)
+    return nil if parsed.size < 2
+
+    case exam_template.cover_fields
+    when 'id_number'
+      Student.find_by(id_number: parsed[1])
+    when 'user_name'
+      Student.find_by('lower(user_name) = ?', parsed[0].downcase)
```

Comment:
If we want to not rely on sql syntax and use activerecord as much as possible it would be better to do:

```ruby
Student.where(Student.arel_table[:user_name].matches(parsed[0])).first # matches is case insensitive
```

OR

```ruby
Student.where(Student.arel_table[:user_name].lower.eq(parsed[0].downcase)).first
```